File: scripts/algorithms/M/Maximum Binary String After Change/Maximum Binary String After Change.py
# Runtime: 128 ms (Top 57.69%) | Memory: 15.4 MB (Top 69.23%)
class Solution:
    def maximumBinaryString(self, binary: str) -> str:
        zero = binary.count('0') # count number of '0'
        zero_idx = binary.index('0') if zero > 0 else 0 # find the index of fist '0' if exists
        one = len(binary) - zero_idx - zero # count number of '1' (not including leading '1's)
        return f"{binary[:zero_idx]}{'1'*(zero-1)}{'0'*min(zero, 1)}{'1'*one}"

File: scripts/algorithms/M/Maximum Binary String After Change/Maximum Binary String After Change.java
// Runtime: 68 ms (Top 32.6%) | Memory: 45.01 MB (Top 8.1%)

class Solution {
    public String maximumBinaryString(String binary) {
        int n = binary.length();
        StringBuffer ans = new StringBuffer("");
        StringBuffer buffer = new StringBuffer("");
        int onesAfter1stZero = 0;
        boolean found1stZero = false;
        for(int i=0;i<n;i++){
            if(found1stZero){
                if(binary.charAt(i)=='0')
                    ans.append('1');
                else
                    buffer.append('1');
            }
            else{
                if(binary.charAt(i)=='0'){
                    found1stZero = true;
                    buffer.append('0');
                }
                else
                    ans.append('1');
            }
        }
        
        ans.append(buffer);
        return ans.toString();
    }
}